mumble: Enable build on Darwin

dwt opened the pull request:
## Things done

- Modernize mumble build, to use lib.cmake* functions to specify the cmake options
- Rename configureFlags to cmakeFlags, as the build had both, but it is actually a cmake build
- Add Install-Phase to copy the built Application and create a wrapper to start mumble from the shell.

There is more work to be done, and I'd appreciate some guidance:
- The app is missing it's icon, no idea why
- There should be some libraries built, and I seem to be missing them? Other builds depend on mumble to get their libraries. This might be a mistake, it might be that the overlay build target is actually the one that builds should depend on, but I don't understand this yet.
- …

This is based on my poco pull request, as poco is a dependency that didn't build on Darwin before, so that update is required. This is also one of the reasons I'm marking this as draft, to get early feedback.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dwt committed:
mumble: rename configureFlags > cmakeFlags

As a reader I first thought that this package was built using automatke,
when in fact it is built using cmake. This rename should help lessen that confusion.

dwt committed:
mumble: Add correct dependency for use-speex option

Support for the speex codec has long since been removed from upstream.

See https://github.com/mumble-voip/mumble/pull/5869 for details.

the use-bundled-speex option now means to use the speex-dsp library
which is packaged separately in nix.

dwt committed:
mumble: darwin: Fix missing version information in finder

Reported upstream at https://github.com/mumble-voip/mumble/issues/6741

dwt commented:
This PR builds on the changes proposed in #383163. I need the darwin build of poco here, so this probably has to stay a draft until that other pull request is merged.

dwt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384691`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>mumble</li>
    <li>murmur</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mumble</li>
  </ul>
</details>

felixsinger commented:
Please rebase on master.

dwt committed:
mumble: Add dependency on nix provided nlohmann_json

dwt committed:
mumble: Add dependency on nix provided microsoft-gsl library

dwt committed:
mumble: enable building on darwin

dwt committed:
mumble: Remove superfluous bundled-opus option

There is no such option anymore, as unbundled opus is now the only
supported option. See upstream d11fd05062f5684b3ffb698eb4cd3130356b6e9a
for details.

dwt committed:
mumble: Remove superfluous bundled-celt option

This option has been removed by upstream for a long time, as nobody uses
that codec anymore.

For details see upstream 4d05018c2e4f1bda48f6244b38f1e7bdc06de808

dwt commented:
Hm, seems that moving the package to pkgs/by-name/ is not as easy as I thought, might need to revert that.

dwt commented:
The builds itself seem fine:

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384691`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>mumble</li>
    <li>murmur</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mumble</li>
  </ul>
</details>

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -61,8 +61,8 @@ let
         cmakeFlags = [
-          "-D g15=OFF"
-          "-D CMAKE_CXX_STANDARD=17" # protobuf >22 requires C++ 17
+          (lib.cmakeBool "g15" false)
```
Same here. Please remove this commit.

dwt commented:
@felixsinger Do you happen to know why the vet script is unhappy with the move of the package to `/pkgs/by-name/`? The code seems straightforward enough, but …

afh commented:
@dwt, while I don't have an overly specific analysis for you as to why the vet check fails, I'd like to point out that mumble is a "package" of packages, i.e. it provides not only mumble, but also at least murmur.
There are several other packages like this, texinfo and boost come to mind. So it might be worthwhile to have a look at the differences how these packages are constructed and "published" in pkgs/top-level/all-packages.

dwt committed:
mumble: use lib helpers for cmake options where it makes them simpler

These are the cases where bools have to be converted to options, so the
helpers help make the code shorter and easier to read.

dwt committed:
mumble: CMAKE_PREFIX_PATH is not required to build the server.

keenanweaver reviewed (commented):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384691`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>quake3demo</li>
  </ul>
</details>

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -141,9 +142,14 @@ let
       env.NIX_CFLAGS_COMPILE = lib.optionalString speechdSupport "-I${speechd-minimal}/include/speech-dispatcher";
 
-      patches = [
+      patches = lib.optionals stdenv.hostPlatform.isDarwin [
```
Remove the lib.optionals condition. That doesn't belong here.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -109,15 +109,15 @@ let
+        (lib.cmakeBool "warnings-as-errors" false) # conversion error workaround
         (lib.cmakeBool "speechd" speechdSupport)
         (lib.cmakeBool "pulseaudio" pulseSupport)
         (lib.cmakeBool "pipewire" pipewireSupport)
```
Please restore the plain options again and remove lib.cmakeBool.

felixsinger commented on pkgs/top-level/all-packages.nix:
```diff
@@ -14375,19 +14375,16 @@ with pkgs;
-    }).mumble;
-
-  mumble_overlay = (callPackages ../applications/networking/mumble {}).overlay;
+    })
```
Let's rather drop this patch from this PR and do this in a dedicated pull request.

dwt commented:
> @dwt, while I don't have an overly specific analysis for you as to why the vet check fails, I'd like to point out that mumble is a "package" of packages, i.e. it provides not only mumble, but also at least murmur. There are several other packages like this, texinfo and boost come to mind. So it might be worthwhile to have a look at the differences how these packages are constructed and "published" in pkgs/top-level/all-packages.

It seems all those packages have not been migrated to pkgs/by-name/ yet. :/

dwt commented:
friendly ping to @felixsinger - could you perhaps take another look? I hope that everything is ready to merge here.

afh commented:
Apologies for potentially having misled you with my comments, @dwt, I must've confused some general clean-up work with preparations for a migration to `pkgs/by-name`.

For some context see:

* My [refactoring work on texinfo](https://github.com/NixOS/nixpkgs/pull/289690)
* and [infinisil's comment](https://github.com/NixOS/nixpkgs/pull/287918#issuecomment-1939243084) regarding package sets in  the by-name tree 

Hope this helps 🙂

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -114,6 +114,7 @@ let
         "-D bundled-opus=OFF"
         "-D bundled-speex=OFF"
         "-D bundle-qt-translations=OFF"
+        # "-D bundled-renamenoise=OFF" # not packaged in nixpkgs yet
```
It is likely that they switch back to upstream. Please remove this commit.

dwt replied:
As far as I understand the mumble source, they have [the rename noise library vendored as a git submodule](https://github.com/mumble-voip/mumble/tree/master/3rdparty), and use that by default.

So I'm not sure I understand your comment about mumble switching back to upstream?

I wanted to document that this flag is available, but not usable yet because the library is not yet packaged in nixpkgs. Why don't you like it?

returntoreality replied:
He means they switched back to upstream rnnoise in mumble master which is in nixpkgs. See https://github.com/mumble-voip/mumble/pull/6607
This also means renamenoise will probably never get packaged as the only user of  (as far as I know) no longer depends on it.

dwt replied:
Good point, I have noted that as a comment to make the next update easier.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -118,7 +116,6 @@ let
         "-D server=OFF"
         "-D bundled-celt=ON"
         "-D bundled-opus=OFF"
-        "-D bundled-speex=OFF"
```
Hm. They still have a build option for it and the code also has hints on speex as well. https://github.com/mumble-voip/mumble/blob/master/docs/dev/build-instructions/cmake_options.md#bundled-speex

dwt replied:
That is really strange. I found [this pull request that explicitly removes support for the speed codec](https://github.com/mumble-voip/mumble/pull/5869). Also the [3rd-party folder](https://github.com/mumble-voip/mumble/tree/master/3rdparty) does not contain the speed codec anymore, but only some speex-dsp library.

Actually, they did drop support for the Speex codec, but kept the flag to mean usage of the speex-dsp for ... something. That is actually a different package in nix and a different dependency. :/

I have added that dependency as a new commit and added the option again.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -66,6 +66,7 @@ let
         cmakeFlags = [
           "-D g15=OFF"
           "-D CMAKE_CXX_STANDARD=17" # protobuf >22 requires C++ 17
+          "-D CMAKE_UNITY_BUILD=ON" # Upstream uses this in their build pipeline to speed up builds
```
But what does this option do?

dwt replied:
I am no C++ expert, but to my understanding this combines several c++ source or header files into bigger compilation units to speed up compilation. I mainly wanted it to make sure that we compile the code as similar to upstream as possible. [Also see the CMAKE Documentation on that feature](https://cmake.org/cmake/help/v3.31/variable/CMAKE_UNITY_BUILD.html)

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -121,7 +121,6 @@ let
         "-D bundled-gsl=OFF"
         # "-D bundled-renamenoise=OFF" # not packaged in nixpkgs yet
         "-D update=OFF"
-        "-D overlay-xcompile=OFF"
```
But these options are for the client? So why removing it?

felixsinger replied:
This commit should be squashed with the follow-up, [mumble: add overlay=False build option as we have our own target for the overlay](https://github.com/NixOS/nixpkgs/pull/384691/commits/90f24d2657f3cba17004a7fce3348c239cb05052), since this commit alone breaks building mumble.

However, by disabling Overlay support its related menu disappears from the Mumble settings window, and I assume that provides easy use of it. So this might be a breaking change. Unless we figure out how we can bring it back with the mumble_overlay package, I'm rather negative about disabling it.

What's your reason for disabling overlay? Is it needed for MacOS that it compiles or just for cleaning up the build options? In case of MacOS, we could disable it specifically for that platform.

dwt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384691`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>murmur</li>
    <li>quake3demo</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mumble</li>
  </ul>
</details>


`ioq3-scion´ was blocked from compilation on darwin from mumble not building on darwin. Now that it does, it fails for an unrelated reason:

```log
/tmp/nix-build-pan-bindings.drv-0/source/cpp/pan.cpp:343:20: error: exception specification in declaration does not match previous declaration
  343 | ListenSockAdapter::ListenSockAdapter(GoHandle handle) noexcept
      |                    ^
/tmp/nix-build-pan-bindings.drv-0/source/include/pan/pan.hpp:305:5: note: previous declaration is here
  305 |     ListenSockAdapter(GoHandle handle);
      |     ^
```

I am pretty sure that mumble is not to blame for this, so I lean towards ignoring this for now and letting someone else fix this. On Linux it builds fine.

dwt commented:
Friendly ping @felixsinger - what do you think about my comments? Do you think we should add anything to this pull request to get it merged?

felixsinger commented:
> Friendly ping @felixsinger - what do you think about my comments? Do you think we should add anything to this pull request to get it merged?

No need to ping. Please keep in mind that people also have a dayjob and other duties to do. I will reply in a bit.

dwt commented:
@felixsinger Do you maybe have any estimate when you might be able to take a look? [I would really like to continue with my pull requests on Quake 3](https://github.com/NixOS/nixpkgs/pull/388021) (which still needs splitting up and reworking) but that depend on Mumble building on Darwin to do anything.

felixsinger commented:
> @felixsinger Do you maybe have any estimate when you might be able to take a look? [I would really like to continue with my pull requests on Quake 3](https://github.com/NixOS/nixpkgs/pull/388021) (which still needs splitting up and reworking) but that depend on Mumble building on Darwin to do anything.

Will work on this today :)

returntoreality reviewed (approved):
Did not test Darwin compile but linux still works
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384691`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>filesender</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>quake3demo</li>
  </ul>
</details>

dwt commented:
rebased, to make merging easier.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -132,7 +132,10 @@ let
         "-D bundled-json=OFF"
         "-D bundled-gsl=OFF"
-        # "-D bundled-renamenoise=OFF" # not packaged in nixpkgs yet
+        # upstream  switched back from renamenoise to rnnoise, which is packaged
+        # https://github.com/mumble-voip/mumble/pull/6607
+        # so switch to that as soon as upstream releases
+        # "-D bundled-renamenoise=OFF" # not packaged in nixpkgs
```
Why? We don't need that documentation. We don't need to reinvent Mumble documentation. That only adds burden to us maintainers. Remove related commits.

dwt replied:
[x] done

dwt commented:
@felixsinger Could you perhaps give an update on where you stand on this pull request? [I would really like to continue to make quake3 work on darwin...](https://github.com/NixOS/nixpkgs/pull/388021)

dwt commented:
Friendly ping @felixsinger - do you already have a timetable when you can take a look at this?

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -146,8 +146,7 @@ let
           (lib.cmakeBool "ice" iceSupport)
         ]
         ++ lib.optionals iceSupport [
-          "-D Ice_HOME=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
-          "-D CMAKE_PREFIX_PATH=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
```
Very strange, when I check this in `nix repl`, I see this output:

```nix
❯ NIXPKGS_ALLOW_BROKEN=1 nix repl .
nix-repl> zeroc-ice
«derivation /nix/store/cf8dmy8ns4xmmbjc2f5nf2brniyl8djk-zeroc-ice-3.7.10.drv»

nix-repl> lib.getDev zeroc-ice
«derivation /nix/store/cf8dmy8ns4xmmbjc2f5nf2brniyl8djk-zeroc-ice-3.7.10.drv»

nix-repl> lib.getLib zeroc-ice 
«derivation /nix/store/cf8dmy8ns4xmmbjc2f5nf2brniyl8djk-zeroc-ice-3.7.10.drv»

nix-repl> "${lib.getLib zeroc-ice}"  
"/nix/store/91x5bar72f7h1zpf8fn4b70j4yh65vkr-zeroc-ice-3.7.10"

nix-repl> "${lib.getDev zeroc-ice}" 
"/nix/store/alc6vhk36yj9vkq5wmnn1p7imdx2sy1m-zeroc-ice-3.7.10-dev"

nix-repl> "${zeroc-ice}"            
"/nix/store/91x5bar72f7h1zpf8fn4b70j4yh65vkr-zeroc-ice-3.7.10"
```

That is quite confusing to me, as I see the same derivation from both `getLib` and `getDev`, yet stringifying them yields different paths. It may be that the unstringified output has confused me? 🤷

I have undone removing the second path from `Ice_HOME`.

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -112,6 +114,7 @@ let
         "-D bundled-opus=OFF"
         "-D bundled-speex=OFF"
         "-D bundle-qt-translations=OFF"
+        "-D bundled-gsl=OFF"
```
I cannot reproduce my thinking from back then. I have checked the source and it is definitely also used in the murmur server. So GSL is definitely needed for the server. I'm adding it there too. I was not able to prove that this is also used for the overlay, as I couldn't identify its source code yet. Will investigate further there.

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -114,6 +116,7 @@ let
         "-D bundled-opus=OFF"
         "-D bundled-speex=OFF"
         "-D bundle-qt-translations=OFF"
+        "-D bundled-json=OFF"
```
[x] done

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -48,6 +49,7 @@ let
 
         nativeBuildInputs = [
           cmake
+          ninja
```
[x] done

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -112,10 +112,10 @@ let
       cmakeFlags = [
         "-D server=OFF"
         "-D bundled-celt=ON"
-        "-D bundled-opus=OFF"
```
[x] done

dwt commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -115,7 +115,6 @@ let
 
       cmakeFlags = [
         "-D server=OFF"
-        "-D bundled-celt=ON"
```
- [x] done

felixsinger commented:
@ofborg build mumble murmur

felixsinger commented:
@ofborg test mumble

dwt commented:
@felixsinger As I'm slowly becoming a package maintainer too, where can I see the result from those @ofborg commands? I thought I should see them in the GitHub jobs section, but I'm not sure I see them there?

afh commented:
Is this what you are looking for?

https://logs.ofborg.org/?key=nixos/nixpkgs.384691&attempt_id=33469be6-697d-4430-9c66-374e3a037130

- The individual GitHub actions link to the ofBorg logs

felixsinger reviewed (changes requested):
I have a few more comments, but it looks pretty good now and even the CI and tests pass. Thank you for your patience :)

bengsparks commented on pkgs/applications/networking/mumble/disable-overlay-build.patch:
`(lib.cmakeBool "overlay" (!stdenv.hostPlatform.isDarwin))` 😉

dwt replied:
@bengsparks what does the additional paren do, I thought it not to be necessary?

bengsparks replied:
I think it's a syntax error without it

dwt replied:
@felixsinger Sadly yes. Even after disabling the overlay in the build options, the script still runs. That is maybe something that I can report as a bug upstream. -> https://github.com/mumble-voip/mumble/issues/6816
        

Disabling the overlay for darwin seems like a smart idea anyway though, as it documents that its build does not (yet) work. I'll add that to the commit that introduces the patch.

dwt commented on pkgs/applications/networking/mumble/fix-plugin-copy.patch:
`options.binary_dir` is actually `$out` on nix. I am not sure why the structure of that folder is different in nix than it is in their build, do you happen to have an idea why that might be? I was conjecturing that this might be some nix preset for cmake to make the build output more compatible with the way the different outputs work in nix? But I have not yet found a way to verify this.

If that is the case, then we can probably forget upstreaming that patch.

dwt replied:
@felixsinger I am sorry, but I seem to lack the knowledge to understand why the output directory for the plugins is set up differently when building in nix than it is in their build server. From observation I can see that it is, but I don't understand (yet) why. Do you have any input on why that might be? Otherwise I would say keep the assumption that it is just a nix thing this is different and try to move on to make it better later?

felixsinger replied:
Let's work on that later. I will try getting access to the MacOS community builder and maybe I can help figuring things out. Though, I have zero knowledge about MacOS. However, we will see later :)

Hartmnt commented:
Hi,

Not sure, if this is the right place at the right time to add this:

I have toyed around with NixOS a little bit myself and noticed that both mumble and mumble-server (aka murmur) are built without the ``-DBUILD_NUMBER`` cmake flag.
This is not a huge problem, but both the client UI as well as the network protocol messages report ``1.5.0`` for both packages. While this does not break anything, it would be nice to be able to know the patch level servers and clients are on.
Since you are refactoring the package structure and hard-coding the source version (``1.5.735``) anyway, would it be possible to use the "build" component of that version string as ``-D BUILD_NUMBER=735``? Is there a utility function to parse semver strings by any chance?

bengsparks commented:
@Hartmnt worst case scenario is string concatting with `lib.majorVersion`, `lib.minorVersion` and `lib.patchVersion`. 
Alternatively I think you can do a string replace with "." to "" (empty space).

dwt commented:
@Hartmnt I have already pulled out multiple smaller changes into their own pull request to make this one a bit lighter. I think your proposal sounds reasonable, but should probably best become it's own pull request, as this one is already open way too long.

felixsinger commented:
@Hartmnt Thank you for reporting it! I've created #409160.

siraben commented:
@dwt please resolve merge conflicts

felixsinger reviewed (approved):
I think this is good to go after a rebase. Thank you!

dwt commented:
Merge Conflicts resolved, and version display also works great.

<img width="155" alt="image" src="https://github.com/user-attachments/assets/af623758-ae17-42a4-ad73-52a63d058202" />

dwt commented:
@siraben Anything else I can provide?

felixsinger reviewed (approved):
awesome. let's get it in.

felixsinger commented:
> 1 package failed to build:
> ioq3-scion

Not sure why that is listed anyway. It only supports Linux platforms as far as I can see.

FliegendeWurst commented:
It would be good to update meta.platforms of `ioq3-scion` then, but that can also be done in another PR

FliegendeWurst commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -97,58 +109,94 @@ let
+        # Can be removed before the next update of Mumble, as that fix was upstreamed
+        # fix version display in MacOS Finder
+        (fetchpatch {
+          url = "https://github.com/mumble-voip/mumble/pull/6742.patch";
```
Please fetch https://github.com/mumble-voip/mumble/commit/fbd21bd422367bed19f801bf278562f567cbb8b7 instead

dwt commented:
@FliegendeWurst That should do it, anything else you see that can be improved?
